### PR TITLE
kick and kick-job commands

### PR DESCRIPTION
**Pull request by marekyggdrasil**
https://github.com/vimukthi-git/beanstalkg/issues/41

### CODE REVIEW

**Comment by vimukthi-git** [comment]
@marekyggdrasil note that you have update the https://github.com/vimukthi-git/beanstalkg/blob/master/architecture/protocol_config.go as well. This specifies how to parse and reply each command.

File: operation/tube_handler.go [hunk 1]
```diff
@@ -81,6 +81,26 @@ func NewTubeHandler(
 					}
 					commands <- c.Copy()
 				}
+			case architecture.KICK:
+				amount := 0
+				bound := strconv.Atoi(c.Params["bound"])
+				for amount < bound {
```
Comment (vimukthi-git) [inline operation/tube_handler.go]: `for amount := 0; amount < bound; amount++`

**Comment by marekyggdrasil** [comment]
Updated, thanks for reminder @vimukthi-git

**Comment by vimukthi-git** [comment]
@marekyggdrasil is this complete now?

**Comment by marekyggdrasil** [comment]
@vimukthi-git ready for review

File: architecture/tube.go [hunk 2]
```diff
@@ -213,3 +214,33 @@ func (tube *Tube) Bury(command *Command) {
 		command.Err = errors.New(NOT_FOUND)
 	}
 }
+
+func (tube *Tube) Kick(command *Command) {
+	amount := 0
+	bound, err := strconv.Atoi(command.Params["bound"])
+	if err != nil {
+		// handle non-integer number of jobs to kick
```
Comment (vimukthi-git) [inline architecture/tube.go]: Log an error here, set the following to `command.Err` 
`command.Err = errors.New(NOT_FOUND)`
and return from function (`NOT_FOUND` is the only error message allowed for kick https://github.com/vimukthi-git/beanstalkg/blob/master/protocol.md), 
It doesn't make sense to process the command afterwards right?

File: architecture/tube.go [hunk 3]
```diff
@@ -213,3 +214,33 @@ func (tube *Tube) Bury(command *Command) {
...
+	}
+	size := tube.buried.Size()
+	if size < bound {
+		bound = size
+	}
+	for amount < bound {
```
Comment (vimukthi-git) [inline architecture/tube.go]: you can use `for amount := 0; amount < bound; amount++ `

File: tests/kick.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+var fivebeans = require('fivebeans');
```
Comment (vimukthi-git) [inline tests/kick.js]: 👍 👍 👍

File: tests/kick_job.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+var fivebeans = require('fivebeans');
+
+var args = process.argv.slice(2);
```
Comment (vimukthi-git) [inline tests/kick_job.js]: 👍 👍 👍

File: tests/failure_worker.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var fivebeans = require('fivebeans');
```
Comment (vimukthi-git) [inline tests/failure_worker.js]: 👍 👍 👍

File: architecture/protocol_test.go [hunk 7]
```diff
@@ -42,4 +42,14 @@ func TestCommand_Parse(t *testing.T) {
 		fmt.Println("Sixth")
 		t.Fail()
 	}
+
+	c = Command{}
```
Comment (vimukthi-git) [inline architecture/protocol_test.go]: 👍 👍 👍

File: architecture/tube.go (hunk 2 above)
Comment (vimukthi-git) [inline architecture/tube.go]: Also I'm thinking there should be a way to configure the command parameter types in `protocol_config.go` and handle the conversion at the command parse time. Would you like to work on that? I think it should not be that hard.

Comment (marekyggdrasil) [inline architecture/tube.go]: For now we have an array in the config of each command specifying command name, I think good idea would be to add an additional array under it that would specify expected type. Then we can implement some validation. Let me know what are your thoughts guys @vimukthi-git and @stephenwithav .

https://github.com/vimukthi-git/beanstalkg/blob/31697b380513b153cd8a57fb66507af423f72bcd/architecture/protocol_config.go#L68

Comment (vimukthi-git) [inline architecture/tube.go]: @marekyggdrasil Your idea makes sense. I was additionally thinking the `Params` array can be made in to a map. But that feels like unnecessary complication, isn't it? 
I will approve and merge this pull req now. I created a separate issue #67 to discuss and work on validation for parameters.

**Pull request merged**